src/ChessBoard.scala

```diff
@@ -0,0 +1,153 @@
+import com.sun.org.apache.xpath.internal.operations.Or
+
+object ChessBoard extends App {
+
+  type Board = Seq[Seq[String]]
+
+  // Takes a number - say 4 (n)
+  // Uses this to create a board which is  4 x 4 (n x n)
+  def printBoard(board: Board): Unit = {
+    // mkString takes elements in a List and combines them with the specified characters
+    // ex. Seq(1, 2, 3).mkString(" ~ ") would output: "1 ~ 2 ~ 3"
+    val boardText = board.map(row => row.mkString(", ")).mkString("\n")
+    println(boardText)
+  }
+
+  def createBoard(n: Int): Board = {
+    // Empty board
+    // Figure out how to create board
+    val board = Seq.fill(n)(Seq.fill(n)("-"))
+    // try to populate first row
+    // Make Pusdo code comments
+    //printSolution(board)
+    board
+  }
+
+  def markXs(board: Board, row: Int, col: Int): Board = {
+
+    val newRow = board(row).updated(col, "X")
+    val newBoard = board.updated(row, newRow)
+
+    newBoard
+
+  }
+
+  def placeQueen(board: Board, row: Int, col: Int): Board = {
+
+    val newRow = board(row).updated(col, "Q")
+    var newBoard = board.updated(row, newRow)
+
+    //Place X's where Q can no longer go
+    //Navigate whole borads
+
+    def isInvalidSquare(rowI: Int, row:Int,  colI:Int, col:Int): Boolean ={
+      checkDiagonals(rowI,row,colI,col) || checkRow(rowI,row,colI,col) || checkCol(rowI,row,colI,col)
+    }
+
+    def checkDiagonals(rowI: Int, row:Int, colI:Int, col:Int): Boolean ={
+      (0 to newRow.size).exists { i =>
+        ((rowI == (row + i) && colI == (col + i)) || (rowI == (row - i) && colI == (col + i)) ||
+          (rowI == (row - i) && colI == (col - i)) || (rowI == (row + i) && colI == (col - i))) && (rowI != row && colI != col)
+      }
+
+    }
+
+    def checkRow(rowI: Int, row:Int, colI:Int, col:Int): Boolean ={
+      rowI == row && colI != col
+    }
+
+    def checkCol(rowI: Int, row:Int, colI:Int, col:Int): Boolean ={
+      rowI != row && colI == col
+    }
+
+    def markInvalidSquares(b: Board): Board = {
+      var markedBoard = b
+      b.zipWithIndex.flatMap { case (rowL, rowI) =>
+        rowL.zipWithIndex.map { case (colL, colI) =>
+          if (isInvalidSquare(rowI, row, colI, col)) {
+            markedBoard = markXs(markedBoard, rowI, colI)
+          }
+        }
+      }
+      markedBoard
+    }
```

Comment:
These functions can be moved outside of the `placeQueen` function block. With the way things are currently, every time `placeQueen` is called all 5 of these functions are redefined again and again. By moving these functions outside of `placeQueen` they will only be defined once.